micall/tests/test_contig_stitcher.py

```diff
@@ -0,0 +1,804 @@
+                        ref_seq=ref_seq,
+                        match_fraction=0.5,
+                        ),
+        GenotypedContig(name='c',
+                        seq='C' * 20 + 'T' * 50,
+                        ref_name='testref',
+                        ref_seq=ref_seq,
+                        match_fraction=0.5,
+                        ),
+        ]
+
+    results = list(stitch_contigs(contigs))
+    assert len(results) == 2
+
+    assert 100 == len(results[0].seq)
+    assert results[0].seq == 'A' * 50 + 'C' * 50
+    assert results[0].query.name == 'left(a)+overlap(a,b)+right(b)'
+
+    assert results[1].query == contigs[2]
+
+
+def test_stitching_of_all_overlapping_contigs_into_one_sequence(exact_aligner):
+    # Scenario: All contigs have some overlapping parts, resulting in one continuous sequence after stitching.
+
+    ref_seq = 'A' * 100 + 'C' * 100 + 'T' * 100
+
+    contigs = [
+        GenotypedContig(name='a',
+                        seq='A' * 50 + 'C' * 20,
+                        ref_name='testref',
+                        ref_seq=ref_seq,
+                        match_fraction=0.5,
+                        ),
+        GenotypedContig(name='b',
+                        seq='A' * 20 + 'C' * 100 + 'T' * 20,
+                        ref_name='testref',
+                        ref_seq=ref_seq,
+                        match_fraction=0.5,
+                        ),
+        GenotypedContig(name='c',
+                        seq='C' * 20 + 'T' * 50,
+                        ref_name='testref',
+                        ref_seq=ref_seq,
+                        match_fraction=0.5,
+                        ),
+        ]
+
+    results = list(stitch_contigs(contigs))
+    assert len(results) == 1
+
+    result = results[0]
+
+    assert 200 == len(result.seq)
+    assert result.seq == 'A' * 50 + 'C' * 100 + 'T' * 50
+    assert result.query.name == 'left(a)+overlap(a,b)+left(right(b))+overlap(left(a)+overlap(a,b)+right(b),c)+right(c)'
+
+
+def test_stitching_with_empty_contigs(exact_aligner):
+    # Scenario: The function is able to handle and ignore empty contigs.
+
+    ref_seq = 'A' * 100
+
+    contigs = [
+        GenotypedContig(name='a',
+                        seq=ref_seq,
+                        ref_name='testref',
+                        ref_seq=ref_seq,
+                        match_fraction=0.5,
+                        ),
+        GenotypedContig(name='b',
+                        seq='',
+                        ref_name='testref',
+                        ref_seq=ref_seq,
+                        match_fraction=0.5,
+                        ),
+        ]
+
+    results = list(stitch_contigs(contigs))
+    assert set(map(lambda x: x.seq, contigs)) \
+        == set(map(lambda x: x.seq, results))
+
+
+def test_stitching_of_identical_contigs(exact_aligner):
+    # Scenario: The function correctly handles and avoids duplication when identical contigs are stitched together.
+
+    ref_seq = 'A' * 100
```

Comment:
Unused variable? Does your editor warn you about unused variables?

Reply:
Yes, a missed unused variable. My editor did not warn about unused variables at the time of writing this code, but now it does as I finally set up elpy and pylint configurations.

Removed this variable.